src/evict/evict_lru.c

```diff
@@ -856,8 +864,11 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 	 * faster than they are being queued.
 	 */
 	if (__evict_queue_empty(queue)) {
-		cache->evict_empty_score = WT_MIN(WT_EVICT_EMPTY_SCORE_MAX,
-		    cache->evict_empty_score + WT_EVICT_EMPTY_SCORE_BUMP);
+		if (__wt_eviction_needed(session, NULL))
+			cache->evict_empty_score =
+			    WT_MIN(WT_EVICT_EMPTY_SCORE_MAX,
+			    cache->evict_empty_score +
+			    WT_EVICT_EMPTY_SCORE_BUMP);
```

Comment:
I believe we get here if the eviction server is woken for another reason. Either a checkpoint or queuing a page for urgent eviction will lead the eviction server to do this. The eviction server uses ``__wt_evict_update_work`` to decide if there is anything to do, wheras ``__wt_eviction_needed`` checks whether the cache is above a trigger value.

The naming could definitely be clearer.